src/python/pants/base/revision.py

```diff
@@ -6,9 +6,13 @@
                         unicode_literals, with_statement)
 
 import re
-from itertools import izip_longest
+from builtins import map, object, str
+from functools import total_ordering
 
+from future.moves.itertools import zip_longest
 
+
+@total_ordering
```

Comment:
This implements all 6 comparison dunders, given `__eq__` and `__lt__`. 

There is a slight performance penalty. If `Revision` is used a lot, I can manually add the other comparisons.